src/parser/jobs/nin/modules/TrickAttackUsage.tsx

```diff
@@ -0,0 +1,126 @@
+import {Trans, Plural} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Downtime from 'parser/core/modules/Downtime'
+import Suggestions, {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const OPTIMAL_GCD_COUNT = 5 // Opener should be Suiton > AE combo > SE before Trick
+
+const MUDRAS: ActionKey[] = [
+	'TEN',
+	'TEN_KASSATSU',
+	'CHI',
+	'CHI_KASSATSU',
+	'JIN',
+	'JIN_KASSATSU',
+]
+
+export class TrickAttackUsage extends Analyser {
+	static override handle = 'taUsage'
+
+	@dependency private data!: Data
+	@dependency private downtime!: Downtime
+	@dependency private suggestions!: Suggestions
+
+	private mudraActions: number[] = []
+
+	private taCasts: number[] = []
+	private lostTime: number = 0
+	private gcdCount: number = 0
+	private castHook?: EventHook<Events['action']>
+
+	override initialise() {
+		this.mudraActions = MUDRAS.map(k => this.data.actions[k].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.castHook = this.addEventHook(playerFilter.type('action'), this.onCast)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.TRICK_ATTACK.id), this.onTrickAttack)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']) {
+		const action = this.data.getAction(event.action)
+		if (event.timestamp >= this.parser.fight.start_time && action && action.onGcd && this.mudraActions.indexOf(action.id) === -1) {
```

Comment:
```suggestion
		if (event.timestamp >= this.parser.fight.start_time && action?.onGcd && !this.mudraActions.includes(action.id)) {
```
or
```suggestion
		if (event.timestamp >= this.parser.fight.start_time && action?.onGcd && this.mudraActions.indexOf(action.id) === -1) {
```

Reply:
Will adjust this, I'm still getting used to ts syntax.